src/tests/Microsoft.PowerFx.Interpreter.Tests.Shared/DependencyTests.cs

```diff
@@ -0,0 +1,223 @@
+        [InlineData("Patch(local, {'Address 1: City':\"test\"}, { 'Account Name' : \"some name\"})", "Read Accounts: address1_city; Write Accounts: name;")]
+        [InlineData("Patch(local, {accountid:GUID(), 'Address 1: City':\"test\"})", "Read Accounts: accountid; Write Accounts: address1_city;")]
+        [InlineData("Patch(local, Table({accountid:GUID(), 'Address 1: City':\"test\"},{accountid:GUID(), 'Account Name':\"test\"}))", "Read Accounts: accountid; Write Accounts: address1_city, name;")]
+        [InlineData("Patch(local, Table({accountid:GUID(), 'Address 1: City':\"test\"},{accountid:GUID(), 'Account Name':\"test\"}),Table({'Address 1: City':\"test\"},{'Address 1: City':\"test\",'Account Name':\"test\"}))", "Read Accounts: accountid, address1_city, name; Write Accounts: address1_city, name;")]
+
+        // Collect and ClearCollect.
+        [InlineData("Collect(local, Table({ 'Account Name' : \"some name\"}))", "Write Accounts: name;")]
+        [InlineData("Collect(local, local)", "Write Accounts: ;")]
+        [InlineData("ClearCollect(local, local)", "Write Accounts: ;")]
+        [InlineData("ClearCollect(local, Table({ 'Account Name' : \"some name\"}))", "Write Accounts: name;")]
+
+        // Inside with.
+        [InlineData("With({r: local}, Filter(r, 'Number of employees' > 0))", "Read Accounts: numberofemployees;")]
+        [InlineData("With({r: local}, LookUp(r, 'Number of employees' > 0))", "Read Accounts: numberofemployees;")]
+
+        // Option set.
+        [InlineData("Filter(local, dayofweek = StartOfWeek.Monday)", "Read Accounts: dayofweek;")]
+
+        [InlineData("Filter(ForAll(local, ThisRecord.numberofemployees), Value < 20)", "Read Accounts: numberofemployees;")]
+
+        // Summarize is special, becuase of ThisGroup.
+        [InlineData("Summarize(local, 'Account Name', Sum(ThisGroup, numberofemployees) As Employees)", "Read Accounts: name, numberofemployees;")]
+        [InlineData("Summarize(local, 'Account Name', Sum(ThisGroup, numberofemployees * 2) As TPrice)", "Read Accounts: name, numberofemployees;")]
+
+        // Join
+        [InlineData("Join(remote As l, local As r, l.contactid = r.contactid, JoinType.Inner, r.name As AccountName)", "Read Contacts: contactid; Read Accounts: contactid, name;")]
+        [InlineData("Join(remote As l, local As r, l.contactid = r.contactid, JoinType.Inner, r.name As AccountName, l.contactnumber As NewContactNumber)", "Read Contacts: contactid, contactnumber; Read Accounts: contactid, name;")]
+
+        // Set
+        [InlineData("Set(numberofemployees, 200)", "Write Accounts: numberofemployees;")]
+        [InlineData("Set('Address 1: City', 'Account Name')", "Read Accounts: name; Write Accounts: address1_city;")]
+        [InlineData("Set('Address 1: City', 'Address 1: City' & \"test\")", "Read Accounts: address1_city; Write Accounts: address1_city;")]
```

Comment:
> [InlineData("Set('Address 1: City', 'Address 1: City' & \"test\")", "Read Accounts: address1_city; Write Accounts: address1_city;")] [](http://example.com/codeflow?start=7&length=133)

Can we add atleast 1 test for all functions that uses Table?

ShowColumns(), RenameColumns(), AddColumn(), DropColumns(), Remove() etc...